manager/allocator/networkallocator/portallocator.go

```diff
@@ -80,15 +80,39 @@ func reconcilePortConfigs(s *api.Service) []*api.PortConfig {
 		return s.Spec.Endpoint.Ports
 	}
 
+	allocatedPorts := make(map[api.PortConfig]*api.PortConfig)
+	for _, portState := range s.Endpoint.Ports {
+		if portState.PublishMode != api.PublishModeIngress {
+			continue
+		}
+
+		allocatedPorts[api.PortConfig{
+			Name:       portState.Name,
+			Protocol:   portState.Protocol,
+			TargetPort: portState.TargetPort,
+		}] = portState
```

Comment:
Suggest splitting out a function that returns a map key for a given `Port`, with a comment explaining why these particular fields are used in the map key.